keras/layers/recurrent.py

```diff
@@ -228,50 +233,61 @@ def __call__(self, inputs, initial_state=None, **kwargs):
         # then add it to the inputs and temporarily
         # modify the input spec to include the state.
         if initial_state is not None:
-            if hasattr(initial_state, '_keras_history'):
-                # Compute the full input spec, including state
-                input_spec = self.input_spec
-                state_spec = self.state_spec
-                if not isinstance(state_spec, list):
-                    state_spec = [state_spec]
-                self.input_spec = [input_spec] + state_spec
-
-                # Compute the full inputs, including state
-                if not isinstance(initial_state, (list, tuple)):
-                    initial_state = [initial_state]
-                inputs = [inputs] + list(initial_state)
-
-                # Perform the call
-                output = super(Recurrent, self).__call__(inputs, **kwargs)
-
-                # Restore original input spec
-                self.input_spec = input_spec
-                return output
-            else:
-                kwargs['initial_state'] = initial_state
+
+            # We need to build the layer so that state_spec exists.
```

Comment:
Currently, `state_spec` is defined in `Recurrent.build`. When, `initial_state` is passed, we need to build the layer so that we can use `state_spec`.